src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -3341,5 +3400,1030 @@ public void FieldConsideredUnassignedIfInitializationViaProperty()
+                // (4,17): warning CS8851: 'A' defines 'Equals' but not 'GetHashCode'
+                //     public bool Equals(A other)
+                Diagnostic(ErrorCode.WRN_RecordEqualsWithoutGetHashCode, "Equals").WithArguments("A").WithLocation(4, 17)
+                );
+        }
+
+        [Fact]
+        public void RecordEquals_19()
+        {
+            var source = @"
+record struct A
+{
+    public static bool Equals(A x) => throw null;
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (2,15): error CS0736: 'A' does not implement interface member 'IEquatable<A>.Equals(A)'. 'A.Equals(A)' cannot implement an interface member because it is static.
+                // record struct A
+                Diagnostic(ErrorCode.ERR_CloseUnimplementedInterfaceMemberStatic, "A").WithArguments("A", "System.IEquatable<A>.Equals(A)", "A.Equals(A)").WithLocation(2, 15),
+                // (4,24): error CS8877: Record member 'A.Equals(A)' may not be static.
+                //     public static bool Equals(A x) => throw null;
+                Diagnostic(ErrorCode.ERR_StaticAPIInRecord, "Equals").WithArguments("A.Equals(A)").WithLocation(4, 24),
+                // (4,24): warning CS8851: 'A' defines 'Equals' but not 'GetHashCode'
+                //     public static bool Equals(A x) => throw null;
+                Diagnostic(ErrorCode.WRN_RecordEqualsWithoutGetHashCode, "Equals").WithArguments("A").WithLocation(4, 24)
+                );
+        }
+
+        [Fact]
+        public void RecordEquals_RecordEqualsInValueType()
+        {
+            var src = @"
+namespace System
+{
+    public class Object
+    {
+        public virtual bool Equals(object x) => throw null;
+        public virtual int GetHashCode() => throw null;
+    }
+    public class Exception { }
+    public class ValueType
+    {
+        public bool Equals(A x) => throw null;
+    }
+    public class Attribute { }
+    public struct Void { }
+    public struct Boolean { }
+    public struct Int32 { }
+    public interface IEquatable<T> { }
+}
+namespace System.Collections.Generic
+{
+    public abstract class EqualityComparer<T>
+    {
+        public static EqualityComparer<T> Default => throw null;
+        public abstract int GetHashCode(T t);
+    }
+}
+public record struct A;
+";
+            var comp = CreateEmptyCompilation(src, parseOptions: TestOptions.RegularPreview);
+
+            comp.VerifyEmitDiagnostics(
+                // warning CS8021: No value for RuntimeMetadataVersion found. No assembly containing System.Object was found nor was a value for RuntimeMetadataVersion specified through options.
+                Diagnostic(ErrorCode.WRN_NoRuntimeMetadataVersion).WithLocation(1, 1)
+                );
+
+            var recordEquals = comp.GetMembers("A.Equals").OfType<SynthesizedRecordEquals>().Single();
```

Comment:
>Single [](start = 93, length = 6)

I am confused, why Single succeeds here?

Reply:
Because of `.OfType<SynthesizedRecordEquals>()`. There is only one "Equals(R)`. The other one is an `Equals(object)` which is `SynthesizedRecordObjEquals`.

---
In reply to: [597114534](https://github.com/dotnet/roslyn/pull/51900#discussion_r597114534) [](ancestors = 597114534)